integration/presto/src/main/java/org/apache/carbondata/presto/CarbonVectorBatch.java

```diff
@@ -95,6 +95,9 @@ private CarbonColumnVectorImpl createDirectStreamReader(int batchSize, DataType
     } else if (dataType == DataTypes.STRING) {
       return new SliceStreamReader(batchSize, field.getDataType(), dictionaryBlock);
     } else if (DataTypes.isDecimal(dataType)) {
+      if (dictionary != null && dataType instanceof DecimalType) {
```

Comment:
If inside DataTypes.isDecimal check means, it is Decimal type. 
So, why this change required ?

Reply:
when the decimal column uses dictionary encoding,  it can't work fine.
the method field.getDataType() will return IntegerType, It is not Decimal type.